src/block_number.rs

```diff
@@ -0,0 +1,127 @@
+use alloy_primitives::U64;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::{fmt, str::FromStr};
+
+/// A block number or tag.
+#[derive(Copy, Clone, Debug, Default, PartialEq, Eq, Hash)]
```

Comment:
This eventually should come from RPC types.

Reply:
Flagging this also exists here: https://github.com/paradigmxyz/reth/blob/529635f8d4d627d3652807e3f0377b3c256de4fc/crates/primitives/src/block.rs#L548.